### PR TITLE
Method assertCmsCategoryExists doesn't return anything, it throws an exception

### LINKED ISSUE #22340: [CO] Unable to change existing CMS Page's category association (EASY FIX)
#### Describe the bug

Impossible to change CMS Category on a CMS Page

#### Why

Regression introduced in 1.7, in eb4cba2, see https://github.com/PrestaShop/PrestaShop/commit/eb4cba22bd5769a16c147626c5ea9dc53a545f4e#r44998939
assertCmsCategoryExists has **no return statement**
still used in if clause asking for **true**

The AbstractCmsPageHandler.assertCmsCategoryExists() never returns a value.
See AddCmsPageHandler implementation (not using if clause).

Either move assertion inside if statement or add return statement to AbstractCmsPageHandler.assertCmsCategoryExists().

https://github.com/PrestaShop/PrestaShop/blob/1c565a28540b55e11b07c69905dbb1014ea162dd/src/Adapter/CMS/Page/CommandHandler/EditCmsPageHandler.php#L82

#### Expected behavior

Should be possible to change CMS Category on a CMS Page

#### Steps to Reproduce

1. Try to reassign category on an existing CMS Page
2. Save, see that it was not persisted

#### Additional information

* PrestaShop version: 1.7.7.0
* PHP version: 7.3

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Method assertCmsCategoryExists doesn't return anything, it throws an exception.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #22340
| How to test?  | Follow ticket instruction

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22407)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

File: src/Adapter/CMS/Page/CommandHandler/EditCmsPageHandler.php [hunk 1]
```diff
@@ -77,9 +77,11 @@ public function handle(EditCmsPageCommand $command)
     private function createCmsFromCommand(EditCmsPageCommand $command)
     {
         $cms = $this->getCmsPageIfExistsById($command->getCmsPageId()->getValue());
-        $cmsCategoryId = null === $command->getCmsPageCategoryId() ?: $command->getCmsPageCategoryId()->getValue();
+        $cmsCategoryId = null !== $command->getCmsPageCategoryId() ? $command->getCmsPageCategoryId()->getValue() : null;
+
+        if (null !== $cmsCategoryId) {
+            $this->assertCmsCategoryExists($cmsCategoryId);
```
Comment (Progi1984) [inline src/Adapter/CMS/Page/CommandHandler/EditCmsPageHandler.php]:
```suggestion
        if (null !== $command->getCmsPageCategoryId()) {
            $this->assertCmsCategoryExists($command->getCmsPageCategoryId()->getValue());
            $cms->id_cms_category = $command->getCmsPageCategoryId()->getValue();
```

**Review by Progi1984 — COMMENTED** [review]
A small change

**Comment by SD1982** [comment]
LGTM Thanks @PierreRambaud !!

File: src/Adapter/CMS/Page/CommandHandler/EditCmsPageHandler.php (hunk 1 above)
Comment (Progi1984) [reply src/Adapter/CMS/Page/CommandHandler/EditCmsPageHandler.php]: Ping @PierreRambaud

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by atomiix** [comment]
Thanks @PierreRambaud & @SD1982

**Comment by marionf** [comment]
@PierreRambaud 
It seems the PR has been merged in develop branch while it's milestoned for 1.7.7.1 and the issue is in 1.7.7.X Kanban as it's a regression of the 1.7.7.0

**Pull request merged**